pos_accounting/src/pool/delta/mod.rs

```diff
@@ -22,6 +22,9 @@ pub struct PoSAccountingDelta<'a> {
 pub struct DeltaMergeUndo {
     pool_data_undo: DeltaDataUndoCollection<H256, PoolData>,
     delegation_data_undo: DeltaDataUndoCollection<H256, DelegationData>,
+    pool_balances_undo: DeltaAmountCollection<H256>,
```

Comment:
I don't think we need collections to undo amounts. The delta itself is enough because you could just reverse the sign.